Library/Homebrew/formula.rb

```diff
@@ -872,39 +872,15 @@ def with_logging(log_type)
     @active_log_type = old_log_type
   end
 
-  # This method can be overridden to provide a plist.
-  # @see https://www.unix.com/man-page/all/5/plist/ Apple's plist(5) man page
-  # <pre>def plist; <<~EOS
-  #  <?xml version="1.0" encoding="UTF-8"?>
-  #  <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
-  #  <plist version="1.0">
-  #  <dict>
-  #    <key>Label</key>
-  #      <string>#{plist_name}</string>
-  #    <key>ProgramArguments</key>
-  #    <array>
-  #      <string>#{opt_bin}/example</string>
-  #      <string>--do-this</string>
-  #    </array>
-  #    <key>RunAtLoad</key>
-  #    <true/>
-  #    <key>KeepAlive</key>
-  #    <true/>
-  #    <key>StandardErrorPath</key>
-  #    <string>/dev/null</string>
-  #    <key>StandardOutPath</key>
-  #    <string>/dev/null</string>
-  #  </dict>
-  #  </plist>
-  #  EOS
-  # end</pre>
-  def plist
-    nil
+  # The generated launchd {.plist} service name.
+  def self.plist_name(label = nil)
+    plist_name = name.split("::").last.gsub(/([a-z])([A-Z])/, '\1-\2').downcase
```

Comment:
This won't work well. The class name cannot be easily reverted, since the original name can contain either `-` or `_`, both of which are stripped, `+` is converted to `x` and `@` is converted to `AT`.

Reply:
> This won't work well.

It feels too hacky anyway…

> The alternative would be to refactor Formulary to load the formula class with the name.

This sounds like the more _correct_ approach… but how exactly do I expose it? From inside `self.load_formula`?

Reply:
Never mind… think I found a way to avoid this altogether!